subsys/usb/usb_device.c

```diff
@@ -143,6 +143,15 @@ static struct usb_dev_priv {
 	bool remote_wakeup;
 } usb_dev;
 
+/** Setup packet definition used to read raw data from USB line */
+struct usb_setup_packet_packed {
+	uint8_t bmRequestType;
+	uint8_t bRequest;
+	uint16_t wValue;
+	uint16_t wIndex;
+	uint16_t wLength;
+} __packed;
```

Comment:
This struct is already naturally aligned, so doesn't need the __packed.

Could you also add a BUILD_ASSET(sizeof(...) == nnn) to double check that the struct matches the wire format?

Reply:
> Could you also add a BUILD_ASSET(sizeof(...) == nnn) to double check that the struct matches the wire format?

nack, that is not necessary. Generally no new ASSERTs in the stack.

Reply:
To confirm: the BUILD_ASSET is to check at compile time that the struct matches the expected wire size.  I'm not worried either way, but I've found these to be useful in the past especially when you're supporting many architectures.

Reply:
>This struct is already naturally aligned, so doesn't need the __packed.

Very much disagree here. Nothing is naturally aligned, it depends on the native types of the CPU. Some CPUs might only be able to address 32-bit at a time and thus this would have padding. Anything that goes on the wire needs to be `__packed`, **always**. I also disagree about adding a `BUILD_ASSERT()`, what is the person that sees it supposed to do? Add `__packed` to the struct.

Reply:
It's all good.  __packed SGTM.